parkbenchcommon/broadcastconsumer.py

```diff
@@ -0,0 +1,112 @@
+# Copyright 2018 Joel Allen Luellwitz and Emily Frost
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+"""Provides the consuming component of a filesystem-based IPC mechanism."""
+
+import datetime
+import os
+import logging
+import time
+
+__all__ = ['BroadcastCheckError', 'BroadcastConsumer']
+
+SPOOL_PATH = '/var/spool'
+
+class BroadcastCheckError(Exception):
+    """This exception is raised when an error is encountered while attempting to check a
+    broadcast.
+    """
+
+class BroadcastConsumer(object):
+    """Provides the consuming component of a filesystem-based IPC mechanism.
+
+    program_name: The name of the program that issues this broadcast.
+    broadcast_name: The name of this broadcast.
+    minimum_delay: The minimum delay in seconds between broadcasts. If a second broadcast
+        occurs before minimum_delay has passed, the second broadcast is ignored.
+    """
+    def __init__(self, program_name, broadcast_name, minimum_delay):
+        self.logger = logging.getLogger(__name__)
+        self.logger.debug("Initializing consumer for broadcast %s from program %s.",
+                          broadcast_name, program_name)
+
+        self.broadcast_name = broadcast_name
+        self.program_name = program_name
+        self.broadcast_path = os.path.join(SPOOL_PATH, program_name, 'ramdisk', 'broadcast')
+        self.last_consumed_broadcast = datetime.datetime.now().isoformat()
+        self.last_read_broadcast = datetime.datetime.now().isoformat()
```

Comment:
This has the consequence that no beacons broadcast when the program first starts will be consumed. This is actually probably the right thing to do but should be documented.